test/test_status.py

```diff
@@ -32,3 +36,49 @@ def test_status(dirtyrepo):
     for filepath, status in git_status.items():
         assert filepath in git_status
         assert status == git_status[filepath]
+
+
+def test_status_untracked_no(dirtyrepo):
+    git_status = dirtyrepo.status(untracked_files="no")
+    not any(status & pygit2.GIT_STATUS_WT_NEW for status in git_status.values())
+
+
+@pytest.mark.parametrize(
+    "untracked_files,expected",
+    [
+        ("no", set()),
+        (
+            "normal",
+            {
+                os.path.join("untracked_dir", ""),
+                "staged_delete_file_modified",
+                os.path.join("subdir", "new_file"),
+                "new_file",
+            },
+        ),
+        (
+            "all",
+            {
+                "new_file",
+                os.path.join("subdir", "new_file"),
+                "staged_delete_file_modified",
+                os.path.join("untracked_dir", "untracked_file"),
+            },
+        ),
+    ],
+)
+def test_status_untracked_normal(dirtyrepostatus, untracked_files, expected):
```

Comment:
I added a new `.zip` archive which is a copy of `dirtyrepo` which also includes a `.gitignore` and an ignored file.